src/compiler/scala/tools/nsc/classpath/ZipAndJarFileLookupFactory.scala

```diff
@@ -195,43 +197,99 @@ final class FileBasedCache[T] {
   private case class Stamp(lastModified: FileTime, fileKey: Object)
```

Comment:
let's document `fileKey` could be `null`

Reply:
I've updated the code to use a sentinel rather than a null.